### ISSUE TITLE
[1.16.5] Server crash sometimes when player joins in area with immersive engineering.

### ISSUE THREAD

**Issue by GeggsLegs** [BluSunrize/ImmersiveEngineering#4935]
### Description of the issue:
Sometimes when a player joins the server with chunks that contain immersive engineering (and sometimes not but that is rare), 
The server will crash. Usually the server detects that a tick takes to long and crashes itself. The player is then not even able to join before getting timed out. This has happened several times before, with one rare instance the last player left the server, after which the server was constantly overloading. After several hours the server crashed with what seemed to be the same crash log.

### Crashlog:
https://gist.github.com/GeggsLegs/dea5355053032492f801abecc1a48621

### Versions & Modlist:
https://gist.github.com/GeggsLegs/749d6abff4f65531b65a8f6b6d515e05

**Comment by GeggsLegs** [comment]
**Some updates:**
I've figured out that this issue occurs sometimes when a player leaves the server and not when joining. When this happens, the server will slowly start using more resources over time, this will eventually cause a crash, a player joining can also cause it to tip over the edge and cause a crash. I used spark to figure if it was really immersive engineering causing this issue
The results are [here](https://spark.lucko.me/C5DvfOQHCu). It seems that about 60% of the servers tick gets used by "java.util.WeakHashMap.getTable()" Which is under immersive engineering. [The memory graph](https://ibb.co/x5v5w5Q) also looks very odd. It looks like a huge chunk of memory is permanently used up, even though there are no players online.
